### ISSUE TITLE
HEAD~1 is no longer treated as a valid ref

### ISSUE THREAD

**Issue by neoeinstein** [bufbuild/buf#3265]
### GitHub repository with your minimal reproducible example

https://github.com/bufbuild/buf/

### Commands

```shell
buf breaking <URL>#ref=HEAD~1
```

### Output

```shell
fatal: invalid refspec 'HEAD~1'
```

### Expected Output

No failures. This worked correctly and as expected in 1.34. Appears related to #3197 and #3215, but with a different appearance.

### Anything else?

Is there an alternate formulation we should be using to check against the prior commit on a branch?

**Comment by emcfarlane** [comment]
This syntax should work. Fixing. An alternative for now is to specify the branch `buf breaking --against '.git#branch=main,ref=HEAD~'`.

**Comment by emcfarlane** [comment]
The fix has been released in [v1.39.0](https://github.com/bufbuild/buf/releases/tag/v1.39.0).
